src/javascript/app_2/App/Components/Routes/__tests__/button_link.spec.js

```diff
@@ -0,0 +1,36 @@
+import React                  from 'react';
+import { expect }             from 'chai';
+import { configure, shallow } from 'enzyme';
+import Adapter                from 'enzyme-adapter-react-16';
+import { Link }               from 'react-router-dom';
+import { ButtonLink }         from '../index';
+
+configure({ adapter: new Adapter() });
+
+describe('<ButtonLink />', () => {
+    it('should render one <ButtonLink /> component', () => {
+        const wrapper = shallow(<ButtonLink />);
+        expect(wrapper).to.have.length(1);
+    });
+    it('should render children when passed in', () => {
+        const child_div = <div className='sweet-child-of-mine' />;
+        const wrapper = shallow(
+            <ButtonLink>
+                { child_div }
+            </ButtonLink>
+        );
+        expect(wrapper.contains(child_div)).to.equal(true);
+    });
+    it('should render one <Link />', () => {
+        const wrapper = shallow(
+            <ButtonLink />
+        );
+        expect(wrapper.find(Link)).to.have.length(1);
+    });
+    it('should render component with className if any given', () => {
+        const wrapper = shallow(
+            <ButtonLink className='a-cool-classname' />
+        );
+        expect(wrapper.find('.a-cool-classname').exists());
+    });
+});
```

Comment:
Nitpick: No newline at the end of file

Reply:
@kellybinary @morteza-binary I just added new line at end of files were it was missing and dismissed approvals, could you please approve it again?